### PR TITLE
CI: add e2e cron job

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding corn job to run master on staging
taking back the config from #14282, just changing the trigger

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-app-cloud-e2e-test.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+name: cloud-e2e-testing
+
+# Used to run the e2e tests on lightning.ai
+on:
+  schedule:
```
Comment (manskx) [inline .github/workflows/ci-app-cloud-e2e-test.yml:5]:
```suggestion
  workflow_dispatch: # TODO: add inputs for specific git_head

  schedule:
```

**Comment by manskx** [comment]
We should figure out how to get notified of failures. We could do:
- A slack notification 
- Automatically block merging PRs if the previous cron run has failed

**Review by manskx — APPROVED** [review]

File: .github/workflows/ci-app-cloud-e2e-test.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-app-cloud-e2e-test.yml:12]: Addressed by trigger on change in this file...

**Comment by Borda** [comment]
> * A slack notification

yes

> * Automatically block merging PRs if the previous cron run has failed

no, does not make sense while this staging is unstable by its nature :rabbit:

**Review by narJH27 — APPROVED** [review]
lgtm!

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by akihironitta — APPROVED** [review]

**Pull request merged**